### DISCUSSION TITLE
Minimum Hardware Requirements

### DISCUSSION THREAD

**Discussion by janstadt** [cagnulein/qdomyos-zwift#893] [General]
This is not a bug but i was curious if you have recommendations for hardware to run QZ on your own hardware or can provide any metrics for the app when its running? I have an original Raspberry Pi (700mhz processor 256mb memory) and im currently on the `make` step of the installation and its taking far longer than 45 minutes which i presume is due to the hardware its running on. Once its all built and running, does the bridge require a lot of processing power or will this old thing be sufficient? I see that the docs state the Raspberry Pi Zero which is a 1ghz processor with 512mb so im hopefully this old pi i am installing it on will have a new purpose!

**Comment by cagnulein** [comment]
hi @janstadt I converted this on a discussion ;)
Yes QZ can run smoothly on a PI pico without any issue, the bridge code is very small and very light!
On the counter side, i don't think that the PI pico can run the QML code, but I didn't try this since a lot of months, so give it a try and be sure to have the GPU of the PI pico enabled! It will impact a lot of the performance of the QML side.

For the make steps, be sure to run the make without any "-jX" thing, because using this option will require a lot of RAM causing the blocks of the system. 

Let me know if you any questions!

**Reply by rbolinger** [reply]
Reading this discussion intrigued me about the possibility of running QZ on a Pico W though I don't even know where to get started. I've run through basic tasks with the Pico such as loading a program and doing some basic MicroPython development. I've also gotten QZ up and running on a Raspberry Pi Zero W. I assume the only things required for this are what is described as the "bridge code" and the settings file though not sure how to decipher what code I need for the bridge or how to run the build the application for the Pico.

Desired goal is to get the Pico W running QZ to connect to the QZ companion app running on a NordicTrack S22i and also have it broadcast to Zwift. I've gotten this working with an Android device running QZ but love the idea of  the Pico W for that due to price and size, much easier to dedicate it to that purpose. Any help to accomplish this is appreciated.

**Comment by cagnulein** [comment]
first of all build it with the instructions. remember to use make without
any -j arguments

than i can help you with the settings

let me know

Il giorno lun 19 giu 2023 alle 21:04 Ryan Bolinger ***@***.***>
ha scritto:

> Reading this discussion intrigued me about the possibility of running QZ
> on a Pico W though I don't even know where to get started. I've run through
> basic tasks with the Pico such as loading a program and doing some basic
> MicroPython development. I've also gotten QZ up and running on a Raspberry
> Pi Zero W. I assume the only things required for this are what is described
> as the "bridge code" and the settings file though not sure how to decipher
> what code I need for the bridge or how to run the build the application for
> the Pico.
>
> Desired goal is to get the Pico W running QZ to connect to the QZ
> companion app running on a NordicTrack S22i and also have it broadcast to
> Zwift. I've gotten this working with an Android device running QZ but love
> the idea of the Pico W for that due to price and size, much easier to
> dedicate it to that purpose. Any help to accomplish this is appreciated.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/893#discussioncomment-6222776>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWGEUFWOR7OCOVLB7VLXMCPFLANCNFSM56HKAGRQ>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud
